File: 0594-longest-harmonious-subsequence/0594-longest-harmonious-subsequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findLHS = function(nums) {
    const freq = new Map();
    for (let num of nums) {
        freq.set(num, (freq.get(num) || 0) + 1);
    }

    let result = 0;
    for (let [num, count] of freq.entries()) {
        if (freq.has(num + 1)) {
            result = Math.max(result, count + freq.get(num + 1));
        }
    }

    return result;
};

File: 0594-longest-harmonious-subsequence/0594-longest-harmonious-subsequence.py
class Solution:
    def findLHS(self, nums: List[int]) -> int:
        freq = defaultdict(int)
        result = 0

        for num in nums:
            freq[num] += 1

        for num in freq:
            if num + 1 in freq:
                result = max(result, freq[num] + freq[num + 1])
        
        return result